drivers/pwm/pwm_mspm0.c

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2024 Bang & Olufsen A/S, Denmark
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_mspm0_pwm
+
+#include <ti/driverlib/dl_timera.h>
+#include <ti/driverlib/dl_timerg.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/logging/log.h>
+
+#define REG_TIMA0            (GPTIMER_Regs *)0x40860000
+#define REG_TIMA1            (GPTIMER_Regs *)0x40862000
+#define IS_TIMA(tim)         (tim == REG_TIMA0 || tim == REG_TIMA1)
+// from the datasheet, only TIMA0 has 4 capture compare channels
+#define HAS_FOUR_CC(tim)     (tim == REG_TIMA0)
+#define MAX_ALLOWED_CHANNELS_TIMX         2
+#define MAX_ALLOWED_CHANNELS_WITH_FOUR_CC 4
+
+LOG_MODULE_REGISTER(pwm_mspm0, CONFIG_PWM_LOG_LEVEL);
+
+struct pwm_mspm0_config {
+	GPTIMER_Regs *timer;
+	const struct pinctrl_dev_config *pinctrl;
+	const uint32_t frequency;
+	const DL_Timer_ClockConfig clock_cfg;
+	const DL_TIMER_PWM_MODE pwm_mode;
+};
+
+static inline int pwm_mspm0_get_cc_direction(int cc_index, int *direction)
```

Comment:
```suggestion
static inline int pwm_mspm0_cc_output_direction(const int cc_index)
```
Renaming to make it more clear that this doesn't get a configuration state of the HW - it merely does a lookup. This could btw be somewhat simplified as just an array that you index.

I'd also drop the output pointer and return the value directly. Cleaner API imo and you can still check for negative return for error